web/next.config.js

```diff
@@ -1,6 +1,25 @@
 /** @type {import('next').NextConfig} */
 const nextConfig = {
-  output: "standalone",
+  distDir: process.env.NODE_ENV === "production" ? "../app" : "./.next",
+  output: process.env.NEXT_PUBLIC_FOR_ELECTRON === "true" && process.env.NODE_ENV === "production" ? "export" : "standalone", // Use standalone output for a server-based Next.js app
+  images: {
+    unoptimized: process.env.NEXT_PUBLIC_FOR_ELECTRON === "true" ? true : undefined,
+  },
+  webpack:(config)=>{
+    config.module.rules.push({
+      test: /\.tsx$/,
+      use: [
+        {
+          loader: 'webpack-preprocessor-loader',
+          options: {
+            params: {
+              isWeb: process.env.NEXT_PUBLIC_FOR_ELECTRON!=="true",
```

Comment:
`isWeb` isn't the clearest name. Maybe `isElectron` and do `#!if !isElectron` in the pages?
```suggestion
              isElectron: process.env.NEXT_PUBLIC_FOR_ELECTRON === "true",
```